File: java/src/main/java/problems201_300/InvertBinaryTree.java
package problems201_300;

import concepts.TreeNode;

enum InvertBinaryTree {
    DEFAULT {
        @Override
        TreeNode invertTree(TreeNode root) {
            if (root == null) {
                return null;
            }
            TreeNode temp = root.left;
            root.left = invertTree(root.right);
            root.right = invertTree(temp);
            return root;
        }
    };

    abstract TreeNode invertTree(TreeNode root);
}

File: kotlin/src/main/kotlin/problems201_300/InvertBinaryTree.kt
package problems201_300

import concepts.TreeNode

enum class InvertBinaryTree {
    DEFAULT {
        override fun invertTree(root: TreeNode?): TreeNode? {
            if (root == null) {
                return null
            }
            val temp = root.left
            root.left = invertTree(root.right)
            root.right = invertTree(temp)
            return root
        }
    },
    ;

    abstract fun invertTree(root: TreeNode?): TreeNode?
}
